include/oneapi/dpl/pstl/memory_impl.h

```diff
@@ -119,10 +119,7 @@ struct __op_uninitialized_copy<_ExecutionPolicy>
     operator()(_SourceT&& __source, _TargetT& __target) const
     {
         using _TargetValueType = std::decay_t<_TargetT>;
-        if constexpr (std::is_trivial_v<_TargetValueType>)
```

Comment:
This code were introduced in https://github.com/oneapi-src/oneDPL/pull/746

Reply:
@kboyarinov, @rarutyun  Here we simple remove the `if constexpr` branch. But also we have variant to use `std::is_arithmetic_v` instead of `std::is_trivial_v`.  What do you think about this?

Reply:
Looking at https://stackoverflow.com/questions/57432081/is-placement-new-required-when-using-allocated-memory-with-fundamental-data-type I think the approach in this PR is correct.

Reply:
My opinion is the following. I don't understand why assigning the arithmetical type is more beneficial then calling the constructor of the same type. For cases like the code above, where know exactly that our type is arithmetical, writing assignment may be more clear to understand comparing with placement new call. We are a library and have to write generic code where the exact type is unknown. Hence we will need to support 2 variants that makes our code more complex with doubtful benefit.

Reply:
An other thing is that the code above implements an assignment, not a constructor and hence assigning the inner integer element is expected operation.

Reply:
You a re ton tight - please take a look to the line below:
```C++
 if (type == Tag::text) s.~string(); // explicit destroy
```